Tools/WebKitTestRunner/TestController.cpp

```diff
@@ -2688,11 +2709,24 @@ void TestController::decidePolicyForNotificationPermissionRequest(WKPageRef page
 void TestController::decidePolicyForNotificationPermissionRequest(WKPageRef, WKSecurityOriginRef origin, WKNotificationPermissionRequestRef request)
 {
     auto originName = originUserVisibleName(origin);
+
+    auto securityOriginString = adoptWK(WKSecurityOriginCopyToString(origin));
+    auto dictionary = adoptWK(m_webNotificationProvider.notificationPermissions());
+    auto value = WKDictionaryGetItemForKey(dictionary.get(), securityOriginString.get());
+    bool permissionDenied = value && !(WKBooleanGetValue(static_cast<WKBooleanRef>(value)));
```

Comment:
Too many parentheses here. Should be:
`bool permissionDenied = value && !WKBooleanGetValue(static_cast<WKBooleanRef>(value));`

Reply:
Fixed (with a slight change since we no longer call `m_webNotificationProvider.notificationPermissions()`